sdk/trace/src/test/scala/org/typelevel/otel4s/sdk/trace/exporter/InMemorySpanExporter.scala

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2023 Typelevel
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.typelevel.otel4s.sdk.trace.exporter
+
+import cats.Applicative
+import cats.data.Chain
+import cats.effect.Concurrent
+import cats.effect.Ref
+import cats.syntax.functor._
+import org.typelevel.otel4s.sdk.trace.data.SpanData
+
+// todo: should be in the testkit package
+final class InMemorySpanExporter[F[_]: Applicative] private (
+    storage: Ref[F, Chain[SpanData]]
```

Comment:
I'm not a huge fan of `Chain` in general, as it (for no apparent reason) does not participate in the stdlib collections framework, or even a custom collections framework within cats.data. it's also unclear how its performance compares to a reasonable equivalent from the stdlib such as `Queue`; it's only compared to `List` and `Vector` in its docs. though it's hard to argue with how well it semantically fits repeated calls to `exportSpans`

Reply:
What if we use Vector instead? For an (upcoming) testkit exporter, it should be enough

Reply:
hmm. are you okay with `Ref[F, Queue[SpanData]]` and having `finishedSpans` return `F[Seq[SpanData]]`? I think [`Queue`](https://www.scala-lang.org/api/current/scala/collection/immutable/Queue.html) fits better than `Vector` here

Reply:
Done